src/rust/id.shinigami/src/lib.rs

```diff
@@ -0,0 +1,287 @@
+    if data.get("retcode").as_int()? != 0 {
+        return Ok(MangaPageResult { manga: Vec::new(), has_more: false });
+    }
+
+    let manga_array = data.get("data").as_array()?;
+    let mut mangas = Vec::new();
+
+    for manga in manga_array {
+        let manga_obj = manga.as_object()?;
+        let title = manga_obj.get("title").as_string()?.read();
+        let _alt_title = manga_obj.get("alternative_title").as_string()?.read();
+        
+        let cover = if let Ok(url) = manga_obj.get("cover_image_url").as_string() {
+            url.read()
+        } else {
+            manga_obj.get("cover_image_url").as_string()?.read()
+        };
+
+        let taxonomy = manga_obj.get("taxonomy").as_object()?;
+        let mut genres = Vec::new();
+        if let Ok(genre_array) = taxonomy.get("Genre").as_array() {
+            for genre in genre_array {
+                if let Ok(genre_obj) = genre.as_object() {
+                    if let Ok(genre_name) = genre_obj.get("name").as_string() {
+                        genres.push(genre_name.read());
+                    }
+                }
+            }
+        }
+
+        // Fix in get_manga_list function
+        mangas.push(Manga {
+            id: manga_obj.get("manga_id").as_string()?.read(),
+            title,
+            author: get_taxonomy_names(&taxonomy, "Author"),
+            artist: get_taxonomy_names(&taxonomy, "Artist"),
+            description: manga_obj.get("description").as_string()?.read(),
+            url: manga_obj.get("manga_id").as_string()?.read(),
+            cover,
+            categories: genres,
+            status: match manga_obj.get("status").as_int()? {
+                1 => MangaStatus::Ongoing,
+                2 => MangaStatus::Completed,
+                _ => MangaStatus::Unknown,
+            },
+            nsfw: MangaContentRating::Safe,
+            viewer: MangaViewer::Rtl
+            // Remove ..Default::default() here
+        });
+    }
+
+    let meta = data.get("meta").as_object()?;
+    let current_page = meta.get("page").as_int()?;
+    let total_pages = meta.get("total_page").as_int()?;
+    
+    Ok(MangaPageResult {
+        manga: mangas,
+        has_more: current_page < total_pages,
+    })
+}
+
+fn get_taxonomy_names(taxonomy: &ObjectRef, key: &str) -> String {
+    if let Ok(array) = taxonomy.get(key).as_array() {
+        let mut names = Vec::new();
+        for item in array {
+            if let Ok(obj) = item.as_object() {
+                if let Ok(name) = obj.get("name").as_string() {
+                    names.push(name.read());
+                }
+            }
+        }
+        names.join(", ")
+    } else {
+        String::new()
+    }
+}
+
+#[get_chapter_list]
+fn get_chapter_list(id: String) -> Result<Vec<Chapter>> {
+    let url = format!(
+        "{}/v1/chapter/{}/list?page=1&page_size=3000&sort_by=chapter_number&sort_order=desc",
+        API_URL, id
+    );
+    let json = Request::new(url, HttpMethod::Get)
```

Comment:
simpler to do this. there are other places to do this as well.
```suggestion
    let json = Request::get(url)
```